File: src/geek time/c#/week03/111. Minimum Depth of Binary Tree.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace week03 {
    public partial class Solution {
        public int MinDepth(TreeNode root) {
            if (root == null) return 0;
            return MinRecursive(root, 1) + 1;
        }

        private int MinRecursive(TreeNode root, int cur) {
            if (root == null) return cur - 1;
            if (root.left == null && root.right == null) return cur - 1;
            var left = int.MaxValue;
            var right = int.MaxValue;
            if (root.left != null)
                left = MinRecursive(root.left, cur + 1);
            if (root.right != null)
                right = MinRecursive(root.right, cur + 1);
            return Math.Min(left, right);
        }
    }
}

File: src/geek time/typescript/week03/111. Minimum Depth of Binary Tree.ts
export class TreeNode {
    public val: number;
    public left: TreeNode;
    public right: TreeNode;
    constructor(val: number) {
        this.val = val;
        this.left = null;
        this.right = null;
    }
}

function minDepth(root: TreeNode | null): number {
    if (!root) return 0;
    return recursiveFind(root, 1) + 1;
}

function recursiveFind(root: TreeNode | null, level: number): number {
    if (!root) return level - 1;
    if (!root.left && !root.right) return level - 1;
    let left = 2147483647;
    if (root.left) left = recursiveFind(root.left, level + 1);
    let right = 2147483647;
    if (root.right) right = recursiveFind(root.right, level + 1);
    return Math.min(left, right);
}
